src/main.rs

```diff
@@ -38,7 +34,8 @@ async fn main() -> Result<(), Box<dyn Error>> {
     df_to_table(ctx.clone(), worker_hosts.clone(), DF_TABLE_NAME).await?; // register table in ctx
 
     let load_balancer = Arc::new(RwLock::new(
```

Comment:
imo, there's a lot of things going on here for this one line. It would make the code clean to separate the initialization of the variable, and then wrap it into atomic reference counter with read/write lock after the variable successfully constructed. 

For example, what if you need to replace `.expect()` into matching block?

Reply:
It's not a bug, it's a feature :)  I wanted to hide complexity of fetching data from db and processing it into datafusion. But again maybe it's better to be more clear in the code.